grails-app/controllers/au/org/ala/biocache/hubs/OccurrenceController.groovy

```diff
@@ -141,9 +174,35 @@ class OccurrenceController {
                         remove = k
                     }
                 }
-                if (remove) searchResults?.activeFacetMap?.remove(remove)
+                if (remove) {
+                    searchResults?.activeFacetMap?.remove(remove)
+                    searchResults?.activeFacetObj?.remove(remove)
+                }
+            }
+
+            Map activeDefaultFilters = [:]
+            Map activeUserFilters = [:]
+            searchResults?.activeFacetObj.each { k, v ->
+                // v is Facet list here, grouped by filtername
+                // fina all user filters
+                def userFilters = v.findAll { !fqToCategoryMap.containsKey(it.displayName) }
```

Comment:
Can you explain what this line does to specifically identify filters that users have added themselves and what it does to ensure that other filters won't be removed from the user interface by this process?